grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -812,4 +816,30 @@ class RequisitionService {
             WHERE r.origin = :location AND r.dateCreated >= :startDate AND r.dateCreated <= :endDate AND r.isTemplate = false 
             """, ['location': location, 'startDate': startDate, 'endDate': endDate])[0] ?: 0
     }
+
+    Event createEvent(EventCode eventCode, Location eventLocation, Date eventDate) {
+        EventType eventType = EventType.findByEventCode(eventCode)
+        if (!eventType) {
+            throw new NotFoundException("No event type with code ${eventCode} has been found")
+        }
+        return new Event(eventDate: eventDate, eventType: eventType, eventLocation: eventLocation)
+    }
+
+    void triggerRequisitionStatusTransition(Requisition requisition, RequisitionStatus status) {
+        if (status == RequisitionStatus.PENDING_APPROVAL) {
+            if (requisition.origin.approvalRequired) {
+                Event event = createEvent(EventCode.PENDING_APPROVAL, requisition.origin, new Date())
+                requisition.addToEvents(event)
+                requisition.status = RequisitionStatus.PENDING_APPROVAL
+                requisition.approvalRequired = true
+                requisition.save()
+                return
+            }
+            Event event = createEvent(EventCode.SUBMITTED, requisition.origin, new Date())
```

Comment:
I am not sure if `SUBMITTED` is a correct code here, so ready for discussion/suggestions.
(It is if we submit a request "in an old way", so if we don't support the request approval workflow)